### PR TITLE
Enabling metrics by user agent

### PR DESCRIPTION
**Pull request by assyadh**
*Issue #, if available:* DOTNET-3203

*Description of changes:* Enabling metrics by user agent when we build the user pool client

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by assyadh** [comment]
My HEAD was wrong on this one, the diff is correct tho. I would fix this upon merge.

File: src/Amazon.AspNetCore.Identity.Cognito/Extensions/CognitoServiceCollectionExtensions.cs [hunk 1]
```diff
@@ -96,8 +98,30 @@ public static CognitoUserPool CreateUserPoolClient(IServiceProvider provider)
...
+            if (args == null || !args.Headers.ContainsKey(UserAgentHeader))
+                return;
+
+            args.Headers[UserAgentHeader] = args.Headers[UserAgentHeader] + " CognitoASPNETCoreIdentityProvider/" + GetAssemblyFileVersion();
+        }
+
+        private static string GetAssemblyFileVersion()
+        {
+            var assembly = typeof(CognitoServiceCollectionExtensions).GetTypeInfo().Assembly;
+            AssemblyFileVersionAttribute attribute = assembly.GetCustomAttribute(typeof(AssemblyFileVersionAttribute)) as AssemblyFileVersionAttribute;
```
Comment (costleya) [inline src/Amazon.AspNetCore.Identity.Cognito/Extensions/CognitoServiceCollectionExtensions.cs]: Can we cache this value o we are not doing reflection constantly? Can we just make it a readonly variable?

Comment (assyadh) [reply src/Amazon.AspNetCore.Identity.Cognito/Extensions/CognitoServiceCollectionExtensions.cs]: Moved it out.

**Comment by assyadh** [comment]
Merging following costleya approval

**Pull request merged**